src/slash_commands/score.rs

```diff
@@ -1,155 +1,110 @@
-                    .kind(CommandOptionType::Integer)
-                    .min_int_value(MIN_FU)
-                    .max_int_value(MAX_FU)
-                    .required(true)
-            })
-            .create_option(|option| {
-                option
-                    .name(HONBAS_OPTION)
-                    .description("Number of honbas (counter sticks)")
-                    .kind(CommandOptionType::Integer)
-                    .min_int_value(MIN_HONBAS)
-                    .max_int_value(MAX_HONBAS)
-                    .required(false)
-            })
-            .create_option(|option| {
-                option
-                    .name(MODE_OPTION)
-                    .description("Calculating mode")
-                    .kind(CommandOptionType::String)
-                    .add_string_choice("Default", DEFAULT_MODE)
-                    .add_string_choice("Loose", LOOSE_MODE)
-                    .add_string_choice("Unlimited", UNLIMITED_MODE)
-                    .required(false)
-            });
+impl From<Mode> for PointsCalculationMode {
+    fn from(value: Mode) -> Self {
+        match value {
+            Mode::Default => PointsCalculationMode::Default,
+            Mode::Loose => PointsCalculationMode::Loose,
+            Mode::Unlimited => PointsCalculationMode::Unlimited,
+        }
     }
+}
 
-    async fn handle(
-        &self,
-        ctx: &Context,
-        command: &ApplicationCommandInteraction,
-        _chombot: &Chombot,
-    ) -> Result<()> {
-        let han = get_int_option(&command.data.options, HAN_OPTION)
-            .ok_or(anyhow!("Missing han value"))?;
-        let fu =
-            get_int_option(&command.data.options, FU_OPTION).ok_or(anyhow!("Missing fu value"))?;
-        let honbas = get_int_option(&command.data.options, HONBAS_OPTION).unwrap_or(DEFAULT_HONBAS);
-        let mode = get_string_option(&command.data.options, MODE_OPTION).unwrap_or(DEFAULT_MODE);
-        let points_calculation_mode = match mode {
-            DEFAULT_MODE => Ok(PointsCalculationMode::Default),
-            LOOSE_MODE => Ok(PointsCalculationMode::Loose),
-            UNLIMITED_MODE => Ok(PointsCalculationMode::Unlimited),
-            _ => Err(anyhow!("Invalid mode: {mode}")),
-        }?;
-
-        let han = Han::new(i32::try_from(han)?);
-        let fu = Fu::new(i32::try_from(fu)?);
-        let honbas = Honbas::new(i32::try_from(honbas)?);
-        let points = Points::from_calculated(points_calculation_mode, han, fu, honbas)?;
-        let embed = create_points_embed(han, fu, honbas, &points);
-
-        command
-            .edit_original_interaction_response(&ctx.http, |response| response.add_embed(embed))
-            .await?;
+/// Calculate the score for given number of han and fu points.
+#[poise::command(slash_command)]
+pub async fn score(
+    ctx: PoiseContext<'_>,
+    #[description = "Number of han points"]
+    #[min = -1600]
+    #[max = 1600]
+    han: i64,
```

Comment:
Can't we have `i32` directly instead of explicitly converting later in the code?

Reply:
we can, actually :+1: